tests/pfcwd/test_pfcwd_function.py

```diff
@@ -732,6 +732,73 @@ def storm_detect_path(self, dut, port, action):
         self.traffic_inst.verify_wd_func(action, self.rx_action, self.tx_action)
         return loganalyzer
 
+    def storm_detect_path_cisco_8000(self, dut, port, action):
+        """
+        Storm detection action and associated verifications
+
+        Args:
+            dut(AnsibleHost) : DUT instance
+            port(string) : DUT port
+            action(string) : PTF test action
+
+        Returns:
+            loganalyzer(Loganalyzer) : instance
+        """
+        restore_time = self.timers['pfc_wd_restore_time_large']
+        detect_time = self.timers['pfc_wd_detect_time']
+
+        loganalyzer = LogAnalyzer(ansible_host=self.dut,
+                                  marker_prefix="pfc_function_storm_detect_{}_port_{}".format(action, port))
+        marker = loganalyzer.init()
+        ignore_file = os.path.join(TEMPLATES_DIR, "ignore_pfc_wd_messages")
+        reg_exp = loganalyzer.parse_regexp_file(src=ignore_file)
+        loganalyzer.ignore_regex.extend(reg_exp)
+        loganalyzer.expect_regex = []
+        loganalyzer.match_regex = []
+
+        if action != "dontcare":
+            start_wd_on_ports(dut, port, restore_time, detect_time, action)
+
+        if not self.pfc_wd['fake_storm']:
+            self.storm_hndle.start_storm()
+
+        if action == "dontcare":
+            start_wd_on_ports(dut, port, restore_time, detect_time, "drop")
+
+        # placing this here to cover all action types. for 'dontcare' action,
+        # wd is started much later after the pfc storm is started
+        if self.pfc_wd['fake_storm']:
+            PfcCmd.set_storm_status(dut, self.queue_oid, "enabled")
+        time.sleep(5)
+
+        # Verify that watchdog is not yet triggered.
+        # test pfcwd functionality on a storm, without traffic.
+        if self.pfc_wd['fake_storm']:
+            loganalyzer.expect_regex.extend([EXPECT_PFC_WD_DETECT_RE + fetch_vendor_specific_diagnosis_re(dut)])
```

Comment:
What's the difference does "self.pfc_wd['fake_storm']" make in this context? What conditions would expect "EXPECT_PFC_WD_DETECT_RE" here?

Reply:
@lipxu : This code is same as the previous code.

Reply:
The function is only used for Cisco here, the code path should be unique, not sure whether we need to keep it as previous ones.

Reply:
Let me confirm if fake_storm is ever set for cisco-8000. If not, I will remove this part.